packages/flutter/lib/src/painting/box_painter.dart

```diff
@@ -186,57 +217,99 @@ class Border {
     assert(bottom != null);
     assert(left != null);
 
-    Paint paint = new Paint();
+    Paint paint = new Paint()
+      ..strokeWidth = 0.0; // used for hairline borders
     Path path;
 
-    // TODO(ianh): Handle hairline border by drawing a single line instead of a wedge
-
-    paint.color = top.color;
-    path = new Path();
-    path.moveTo(rect.left, rect.top);
-    path.lineTo(rect.left + left.width, rect.top + top.width);
-    path.lineTo(rect.right - right.width, rect.top + top.width);
-    path.lineTo(rect.right, rect.top);
-    path.close();
-    canvas.drawPath(path, paint);
-
-    paint.color = right.color;
-    path = new Path();
-    path.moveTo(rect.right, rect.top);
-    path.lineTo(rect.right - right.width, rect.top + top.width);
-    path.lineTo(rect.right - right.width, rect.bottom - bottom.width);
-    path.lineTo(rect.right, rect.bottom);
-    path.close();
-    canvas.drawPath(path, paint);
-
-    paint.color = bottom.color;
-    path = new Path();
-    path.moveTo(rect.right, rect.bottom);
-    path.lineTo(rect.right - right.width, rect.bottom - bottom.width);
-    path.lineTo(rect.left + left.width, rect.bottom - bottom.width);
-    path.lineTo(rect.left, rect.bottom);
-    path.close();
-    canvas.drawPath(path, paint);
-
-    paint.color = left.color;
-    path = new Path();
-    path.moveTo(rect.left, rect.bottom);
-    path.lineTo(rect.left + left.width, rect.bottom - bottom.width);
-    path.lineTo(rect.left + left.width, rect.top + top.width);
-    path.lineTo(rect.left, rect.top);
-    path.close();
-    canvas.drawPath(path, paint);
+    switch (top.style) {
```

Comment:
I probably would have written this as an if.  It's not like someone is going to miss this when adding a new border style.  By contrast, the lerp might be a good place to use a switch because you want to draw the developer's attention there.